fleet_vehicle_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Dixmit
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    fleet_vehicle_id = fields.Many2one("fleet.vehicle", string="Vehicle")
+
+
+class PurchaseOrderLine(models.Model):
+
+    _inherit = "purchase.order.line"
+    fleet_vehicle_id = fields.Many2one(
+        comodel_name="fleet.vehicle",
+        compute="_compute_fleet_vehicle_id",
+        store=True,
+        readonly=False,
+    )
+    fleet_vehicle_from_po = fields.Boolean(
+        compute="_compute_fleet_vehicle_from_po", store=True
+    )
+
+    @api.depends("fleet_vehicle_from_po")
+    def _compute_fleet_vehicle_id(self):
+        for rec in self:
+            if rec.fleet_vehicle_from_po:
```

Comment:
I don't understand this boolean field. It sets to `True` when the order is related to a fleet vehicle. So this if section will always be `True` in this case.
So when the order is not related to a fleet vehicle, it'll fill the `fleet_vehicle_id` is an empty recordset (and that's also the expected behavior).

Reply:
This is to cover a practical case where we want to order pieces for several fleet vehicle in one PO:

- If we put the vehicle on the PO, it will be propagate on all purchase order lines and the vehicle will be readonly in the view. In this case the PO is for one vehicle.
- If we don't put the vehicle on the PO, we can order pieces for different vehicles in one PO, changing the vehicle on each line. This allows a big customer to order pieces for more than one vehicle in one PO.